### DISCUSSION TITLE
Does the database related info support IP address

### DISCUSSION THREAD

**Discussion by SimonLiuhd** [apache/skywalking#7895] [Q&A]
Hi all，

    Can the agent collects database info by IP address?
I have the same system deployed in all the data-center，and the database hosts share the same name (such as db1) and the same port，but their IP addresses are different in different data centers (In fact the database host names are mapping in OS hosts files）:
   For instances：
1> in the hosts file of data-center A:  "172.168.1.AAA  db1"
2> 1> in the hosts file of data-center B:  "172.168.1.BBB  db1"
    So the service_traffic table which has the mete-data for databases merges all the data source into only one: **db1:db1:3306**, and we cannot difference them in different data-centers.

So, is there solution for the above case to help us difference the different data-sources? such as use the ip-address for the databases?

Simon

**Comment by wu-sheng** [comment]
https://github.com/apache/skywalking/blob/8579572be26f5a2ea45567e669c788a3dcfb3044/oap-server/analyzer/agent-analyzer/src/main/java/org/apache/skywalking/oap/server/analyzer/provider/trace/parser/listener/MultiScopesAnalysisListener.java#L169

This is how we set up database instance name. Because, typically, in production environment, it is hardly to see this kind of hybrid deployment, considering the importance of database performance.

**Reply by SimonLiuhd** [reply]
![image](https://user-images.githubusercontent.com/89838158/136730136-ad4c4be2-f2de-4381-9f11-ff0c27e49e97.png)
But how we can change the above to show IP address?

**Reply by wu-sheng** [reply]
I have shown you how. The codes are about this.
